services/wireguard/endpoint/userspace/network_windows.go

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package userspace
+
+import (
+	"net"
+	"os/exec"
+	"strconv"
+
+	"github.com/jackpal/gateway"
+	"github.com/pkg/errors"
+)
+
+func assignIP(iface string, subnet net.IPNet) error {
+	out, err := exec.Command("powershell", "-Command", "netsh interface ip set address name=\""+iface+"\" source=static "+subnet.String()).CombinedOutput()
+	return errors.Wrap(err, string(out))
+}
+
+func renameInterface(name, newname string) error {
+	out, err := exec.Command("powershell", "-Command", "netsh interface set interface name=\""+name+"\" newname=\""+newname+"\"").CombinedOutput()
+	return errors.Wrap(err, string(out))
+}
+
+func excludeRoute(ip net.IP) error {
+	gw, err := gateway.DiscoverGateway()
+	if err != nil {
+		return err
+	}
+
+	out, err := exec.Command("powershell", "-Command", "route add "+ip.String()+"/32 "+gw.String()).CombinedOutput()
+	return errors.Wrap(err, string(out))
+}
+
+func addDefaultRoute(name string) error {
+	id, gw, err := interfaceInfo(name)
+	if err != nil {
+		return errors.Wrap(err, "failed to get interfaces info")
+	}
+
+	if out, err := exec.Command("powershell", "-Command", "route add 0.0.0.0/1 "+gw+" if "+id).CombinedOutput(); err != nil {
+		return errors.Wrap(err, string(out))
+	}
+
+	out, err := exec.Command("powershell", "-Command", "route add 128.0.0.0/1 "+gw+" if "+id).CombinedOutput()
+	return errors.Wrap(err, string(out))
+}
+
+func destroyDevice(name string) error {
+	return nil
+}
+
+func interfaceInfo(name string) (id, gw string, err error) {
+	iface, err := net.InterfaceByName(name)
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to get interfaces "+name)
+	}
+
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to get interfaces addresses")
+	}
+
+	var ipv4 net.IP
+	for _, addr := range addrs {
+		ip, _, _ := net.ParseCIDR(addr.String())
```

Comment:
Not a fan of hiding the error here. While I'm pretty sure this will work 99.99999% of the time, I can imagine that if we ever hit a case where this does not work as expected, we'll bang our heads here for quite some time. Log?

Reply:
👍 especially when parsing from such generic type as string

Reply:
We convert to string here from well defined address type here.. but well, if we hit go stdlib bug, it might help :)

Reply:
Added logging just in case. :)